.github/workflows/build.yaml

```diff
@@ -82,68 +111,56 @@ jobs:
 
 
   unittest:
+    needs: build-and-push-image
     runs-on: ubuntu-latest
+    container: 
+      image: ghcr.io/aiven/myhoard-test:${{ github.sha }}
     strategy:
+      fail-fast: false
+      # this isn't a standard matrix because some version testing doesn't make sense
+      # e.g. you cannot use mysql .28 with percona .27, and there's no clean way of
+      # skipping matrix items that aren't meaningful
```

Comment:
This should be possible: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstrategymatrixexclude

Reply:
Technically, it could be possible. But would it be better? Then you need to calculate what's being executed and what not. This approach just uses a "flat list" instead of a matrix, and I doubt we'd need to check all possible python versions (mysql vs percona versions seem fare more risky!)